src/object/srv_ec_aggregate.c

```diff
@@ -2307,6 +2306,7 @@ agg_iterate_pre_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 	}
 
 	if (rc < 0) {
+		agg_clear_extents(agg_entry);
```

Comment:
Shouldn't such cleanup function called in cont_ec_aggregate_cb()? what if vos_iterate() abort on other errors (error not returned by pre callback)?